src/main/java/org/opensearch/neuralsearch/processor/NormalizationProcessor.java

```
@@ -0,0 +1,204 @@
+    final ScoreNormalizationTechnique normalizationTechnique;
+    @Getter(AccessLevel.PACKAGE)
+    final ScoreCombinationTechnique combinationTechnique;
+
+    /**
+     * Need all args constructor to validate parameters and fail fast
+     * @param tag
+     * @param description
+     * @param normalizationTechnique
+     * @param combinationTechnique
+     */
+    public NormalizationProcessor(
+        final String tag,
+        final String description,
+        final String normalizationTechnique,
+        final String combinationTechnique
+    ) {
+        this.tag = tag;
+        this.description = description;
+        validateParameters(normalizationTechnique, combinationTechnique);
+        this.normalizationTechnique = ScoreNormalizationTechnique.valueOf(normalizationTechnique);
+        this.combinationTechnique = ScoreCombinationTechnique.valueOf(combinationTechnique);
+    }
+
+    /**
+     * Method abstracts functional aspect of score normalization and score combination. Exact methods for each processing stage
+     * are set as part of class constructor
+     * @param searchPhaseResult {@link SearchPhaseResults} DTO that has query search results. Results will be mutated as part of this method execution
+     * @param searchPhaseContext {@link SearchContext}
+     * @param <Result>
+     */
+    @Override
+    public <Result extends SearchPhaseResult> void process(
+        final SearchPhaseResults<Result> searchPhaseResult,
+        final SearchPhaseContext searchPhaseContext
+    ) {
+        if (searchPhaseResult instanceof QueryPhaseResultConsumer) {
+            QueryPhaseResultConsumer queryPhaseResultConsumer = (QueryPhaseResultConsumer) searchPhaseResult;
+            Optional<SearchPhaseResult> maybeResult = queryPhaseResultConsumer.getAtomicArray()
+                .asList()
+                .stream()
+                .filter(Objects::nonNull)
+                .findFirst();
+            if (isNotHybridQuery(maybeResult)) {
```

we should make the condition as rather than making it negation.

`if(hybridQueryResults(maybeResults))`

---

Negation is needed to fail fast. Options are to negate in the if clause (```if (!hybridQueryResults(maybeResults))```) which is hard to read and lass error prone, or encapsulate that in the method.